examples/plot_picard_ode.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""
+Anderson acceleration in application to Picard–Lindelöf theorem.
+================================================================
+
+With `Picard–Lindelöf theorem
+<https://en.wikipedia.org/wiki/Picard%E2%80%93Lindel%C3%B6f_theorem>`_ we can reduce differential equation solving to fixed point computations and simple integration.
+More precisely consider the ODE:
+
+.. math::
+
+  y'(t)=f(t,y(t))
+
+of some time-dependant dynamic :math:`f:\mathbb{R}\times\mathbb{R}^d\rightarrow\mathbb{R}^d` and initial conditions :math:`y(0)=y_0`. Then we have:
+
+.. math::
+
+  y(t)=y_0+\int_0^t f(s,y(s))\mathrm{d}s
+
+Then we can define the sequence of functions :math:`(\phi_k)` with :math:`\phi_0=0` recursively as follow:
+
+.. math::
+
+  \phi_{k+1}(t)=y_0+\int_0^t f(s,\phi_k(s))\mathrm{d}s
```

Comment:
To connect with the section on fixed points, it would be great to use the letter `T` (i.e., make it completely obvious what is `fixed_point_fun` in this)